ignition_ros2_control/src/ignition_ros2_control_plugin.cpp

```diff
@@ -0,0 +1,400 @@
+    while (!parameters_client->wait_for_service(0.5s)) {
+      if (!rclcpp::ok()) {
+        RCLCPP_ERROR(
+          node->get_logger(), "Interrupted while waiting for %s service. Exiting.",
+          robot_description_node_.c_str());
+        return 0;
+      }
+      RCLCPP_ERROR(
+        node->get_logger(), "%s service not available, waiting again...",
+        robot_description_node_.c_str());
+    }
+
+    RCLCPP_INFO(
+      node->get_logger(), "connected to service!! %s asking for %s",
+      robot_description_node_.c_str(),
+      param_name.c_str());
+
+    // search and wait for robot_description on param server
+    while (urdf_string.empty()) {
+      std::string search_param_name;
+      RCLCPP_DEBUG(node->get_logger(), "param_name %s", param_name.c_str());
+
+      try {
+        auto f = parameters_client->get_parameters({param_name});
+        f.wait();
+        std::vector<rclcpp::Parameter> values = f.get();
+        urdf_string = values[0].as_string();
+      } catch (const std::exception & e) {
+        RCLCPP_ERROR(node->get_logger(), "%s", e.what());
+      }
+
+      if (!urdf_string.empty()) {
+        break;
+      } else {
+        RCLCPP_ERROR(
+          node->get_logger(), "ignition_ros2_control plugin is waiting for model"
+          " URDF in parameter [%s] on the ROS param server.", search_param_name.c_str());
+      }
+      usleep(100000);
+    }
+    RCLCPP_INFO(
+      node->get_logger(), "Recieved urdf from param server, parsing...");
+
+    return urdf_string;
+  }
+
+  //////////////////////////////////////////////////
+
+public:
+  std::map<std::string, ignition::gazebo::Entity> GetEnabledJoints(
+    const ignition::gazebo::Entity & _entity,
+    ignition::gazebo::EntityComponentManager & _ecm) const
+  {
+    std::map<std::string, ignition::gazebo::Entity> output;
+
+    std::vector<std::string> enabledJoints;
+
+    // Get all available joints
+    std::vector<ignition::gazebo::Entity> jointEntities;
+    jointEntities = _ecm.ChildrenByComponents(_entity, ignition::gazebo::components::Joint());
+
+    // Iterate over all joints and verify whether they can be enabled or not
+    for (const auto & jointEntity : jointEntities) {
+      const auto jointName = _ecm.Component<ignition::gazebo::components::Name>(
+        jointEntity)->Data();
+
+      // Make sure the joint type is supported, i.e. it has exactly one
+      // actuated axis
+      const auto * jointType = _ecm.Component<ignition::gazebo::components::JointType>(jointEntity);
+      switch (jointType->Data()) {
+        case sdf::JointType::PRISMATIC:
+        case sdf::JointType::REVOLUTE:
+        case sdf::JointType::CONTINUOUS:
+        case sdf::JointType::GEARBOX:
+          {
+            // Supported joint type
+            break;
+          }
+        case sdf::JointType::FIXED:
+          {
+            igndbg << "[ignition_ros2_control] Fixed joint [" << jointName <<
+              "(Entity=" << jointEntity << ")] is skipped.\n";
```

Comment:
This plugin is using a combination of Ignition (`igndbg`...) and ROS (`RCLCPP_INFO`...) console logging. It's easier for the user if they only need to check one log. I'd recommend going with ROS logging.